web-registry/src/pages/BatchDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Box } from '@mui/material';
+import { useParams } from 'react-router-dom';
+
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import Section from 'web-components/lib/components/section';
+import Title from 'web-components/lib/components/title';
+
+import { getBatchWithSupplyForDenom } from '../lib/ecocredit';
+import { getMetadata } from '../lib/metadata-graph';
+import { useBatchDetailsQuery } from '../generated/graphql';
+
+import type { BatchInfoWithSupply } from '../types/ledger/ecocredit';
+import type { BatchMetadataLD } from '../generated/json-ld';
+import {
+  BatchInfoGrid,
+  BatchMetadata,
+  BatchTotalsGrid,
+} from '../components/molecules';
+
+export const BatchDetails: React.FC = () => {
+  const { batchDenom } = useParams();
+  const [batch, setBatch] = useState<BatchInfoWithSupply>();
+  const [metadata, setMetadata] = useState<BatchMetadataLD>();
+
+  useEffect(() => {
+    const fetch = async (): Promise<void> => {
+      if (batchDenom) {
+        try {
+          const batch = await getBatchWithSupplyForDenom(batchDenom);
+          setBatch(batch);
+          if (batch.metadata) {
+            const { data: metadata } = await getMetadata(batch.metadata);
+            setMetadata(metadata);
+          }
+        } catch (err) {
+          console.error(err); // eslint-disable-line no-console
+        }
+      }
+    };
+    fetch();
+  }, [batchDenom]);
+
+  const { data: offchainData } = useBatchDetailsQuery({
+    skip: !batchDenom,
+    variables: { batchDenom: batchDenom as string },
+  });
+  const projectHandle =
+    offchainData?.creditVintageByBatchDenom?.projectByProjectId?.handle || '';
+
+  if (!batch) return null;
+  // TODO should there be a not found component or redirect?
```

Comment:
cc/ @erikalogie we generally don't have such thing implemented (for projects, credit classes, and here credit batches), maybe something to think about, either we could just display the existing "not found" page or some new component